BrainPortal/lib/neurohub_helpers.rb

```diff
@@ -145,4 +145,16 @@ def neurohub_search(token, limit=20, user=current_user)
     return report
   end
 
+  # filter NeuroHub messages (no invites)
+  def find_nh_messages(user=current_user)
+    Message.where(:user_id        => user.available_users.map(&:id),
+                  :type           => nil  #  filter out invites
+                  ).order( "last_sent DESC" )
+  end
+
+  def find_nh_contacts(user)
```

Comment:
This is totally not efficient; please don't reconstruction a bunch of objects with all their attributes just to extract their ids, given that rails already provide methods to get the association IDs or pluck individual attribute without such reconstruction.

Reply:
Also document what this method does. I am not even sure what it does by looking at the code. And why in the controller code every time it is invoked, some adjustments are always made to the results

Reply:
Consider running your statements in the console pieces by pieces to see how inefficient they are.